### ISSUE TITLE
WebElement.findElements() not finding child element on Appium Desktop 1.18.3

### ISSUE THREAD

**Issue by mandutree** [appium/appium-desktop#1601]
Unclear if this is Appium or Appium Desktop issue.  However, I was able to reproduce the problem with Appium Desktop version 1.18.3, but the problem does not exist in version 1.17.1-1.

## The problem

The following code describes the problem.
```
val parentLocator = MobileBy.AccessibilityId("parentLocatorId")
val childLocator = By.xpath("//child/xpath/locator")

/* Scroll until parent visible but child not visible on screen */

val parent = driver.findElement(parentLocator)                          // parent found
val child = parent.findElements<WebElement>(childLocator)               // child not found (expected)
val childElements = driver.findElements(childLocator)                   // 2 child items found on page (expected)

/* Scroll until both parent and child visible on screen */

val parentAgain = driver.findElement(parentLocator)                     // parent found
val childAgain = parentAgain.findElements<WebElement>(childLocator)     // child not found (unexpected)
val childElementsAgain = driver.findElements(childLocator)              // 3 child items found on page (expected)
```

In my case, I have a table with multiple rows, and each row containing labels.  The row is the parent element while the label is the child element.  The parent row element has a unique global locator while the child label element has a unique relative locator, but not unique global locator.
It looks as though executing `driver.findElement(parentLocator)`, the first time, creates a cache of element contents and subsequent call reuses part of the cache.
In any case the outcome is; looking for child element within the parent element, the second time, does not find the child element.

## Environment

- I am running Appium Desktop version 1.18.3
- I am on (pick one):
    - [ X] Mac
    - [ ] Windows
    - [ ] Linux

**Comment by KazuCocoa** [comment]
What was each page source result?
Btw, what version of OS did you use?

**Comment by mandutree** [comment]
The mac OS version was 10.15.7

As for the page source, it looks something like this
```
<android.view.ViewGroup>
    <android.widget.Button id="globalUniqueParentId1">
        <android.widget.TextView id="child-label">
        <android.widget.TextView id="child-value">
<android.view.ViewGroup>
    <android.widget.Button id="globalUniqueParentId2">
        <android.widget.TextView id="child-label">
        <android.widget.TextView id="child-value">
<android.view.ViewGroup>
    <android.widget.Button id="globalUniqueParentId3">
        <android.widget.TextView id="child-label">
        <android.widget.TextView id="child-value">
```
To get 3rd child label, you can do something like this: `driver.findElement(By.id("globalUniqueParentId3")).findElements(By.xpath("//android.widget.TextView[@id='child-label']"))`. 
The first time you call this, your screen shows up to `<android.widget.Button id="globalUniqueParentId3">` but not the last 2 lines.  In this case, the above call return 0 elements.  Then you scroll down to make everything visible on screen.  Then make the same call.  At this point, I expect the call to return a WebElement of `<android.widget.TextView id="child-label">` representing the 3rd child label, but it doesn't.  It returns 0 elements.
PS.  ignore any syntax error.  I'm just writing this on text editor.

**Comment by KazuCocoa** [comment]
Perhaps https://github.com/appium/appium-uiautomator2-server/pull/363 or https://github.com/appium/appium-uiautomator2-server/pull/371 is related to.

Could you try out the latest beta or appium 1.19.0 rc? It has a few improvements

**Comment by mandutree** [comment]
It looks like the problem has been fixed on version 1.19.0-beta.0.  I will go ahead and close this ticket.

**Comment by KazuCocoa** [comment]
Good to know.
1.19.0 release will have changes in the current beta which are related to your issue.
